generator/src/main/java/com/scottlogic/deg/generator/walker/reductive/ReductiveState.java

```diff
@@ -4,108 +4,103 @@
+        return fixedFields.containsKey(field);
     }
 
     //get a stream of all possible values for the field that was fixed on the last iteration
-    public Stream<Object> getValuesFromLastFixedField(){
-        if (this.lastFixedField == null)
+    public Stream<Object> getValuesFromNextFieldToFix() {
+        if (this.nextFieldToFix == null)
             throw new NullPointerException("Field has not been fixed yet");
 
-        return this.lastFixedField.getStream();
-    }
-
-    //get a copy of the current fixed field for the given field, will return null if the field isn't fixed
-    public FixedField getFixedField(Field field) {
-        if (lastFixedField != null && lastFixedField.getField().equals(field)){
-            return lastFixedField;
-        }
-
-        return this.fixedFields.getOrDefault(field, null);
+        return this.nextFieldToFix.getStream();
     }
 
     @Override
     public String toString() {
         return toString(false);
     }
 
-    public Map<Field, FixedField> getFixedFieldsExceptLast(){
+    public Map<Field, FixedField> getFixedFields() {
         return this.fixedFields;
     }
 
-    public FixedField getLastFixedField(){
-        return this.lastFixedField;
-    }
-
-    public Set<Field> getUnfixedFields(){
-        return this.fields.stream()
-            .filter(f -> !this.fixedFields.containsKey(f) && !f.equals(this.lastFixedField.getField()))
-            .collect(Collectors.toSet());
+    public FixedField getNextFieldToFix() {
+        return this.nextFieldToFix;
     }
 
     public String toString(boolean detailAllFields) {
         String fixedFieldsString = this.fixedFields.size() > 10 && !detailAllFields
             ? String.format("Fixed fields: %d of %d", this.fixedFields.size(), this.fields.size())
             : String.join(", ", this.fixedFields.values()
-                .stream()
-                .sorted(Comparator.comparing(ff -> ff.getField().name))
-                .map(FixedField::toString)
-                .collect(Collectors.toList()));
+            .stream()
+            .sorted(Comparator.comparing(ff -> ff.getField().name))
+            .map(FixedField::toString)
+            .collect(Collectors.toList()));
 
-        if (this.lastFixedField == null) {
+        if (this.nextFieldToFix == null) {
             return fixedFieldsString;
         }
 
         return this.fixedFields.isEmpty()
-            ? this.lastFixedField.toString()
-            : this.lastFixedField.toString() + " & " + fixedFieldsString;
+            ? this.nextFieldToFix.toString()
+            : this.nextFieldToFix.toString() + " & " + fixedFieldsString;
     }
 
     public ProfileFields getFields() {
         return this.fields;
     }
 
-    public ReductiveState with(FixedField fixedField){
+    public ReductiveState withNextFieldToFixChosen(FixedField nextFieldToFix) {
+        if (this.nextFieldToFix != null) {
+            throw new UnsupportedOperationException();
```

Comment:
Can we add a message here, so it this event happens then there is a supportive message to say what has gone wrong. Doesn't need to be too detailed, but something to support us if we see it happen.

Also the `OperationNotSupportedException` could be a better type to throw?